doc/doxygen/src/Examples.dox

```diff
@@ -255,6 +287,10 @@
  *     # TextFileSpecification
  * )
  *
+ * # Ensure that the manager is capable of the required workflows.
+ * if not (manager.hasCapability(Manager.Capability.kResolution) and
+ *         manager.hasCapability(Manager.Capability.kPublishing)):
+ *   return
```

Comment:
[Fix] The thumbnails sketch needs updating similarly.

Reply:
The thumbnail sketch appears to follow on from the previous example, i.e. `managementPolicy` as assumed to already have been called, so we reasoned that `hasCapability` is similarly assumed to have already been called.

Reply:
I think its meant to be follow on from "some other code" rather than the one before, but as noted, the other one shouldn't be aborting publishing if resolve isn't supported.

Reply:
What do you think of? https://github.com/OpenAssetIO/OpenAssetIO/pull/1141/commits/1dc973e36afe14f4900fda1c04252db9c8354273

We still show capability being a less fine-comb up front precondition for `publishing`, but leave a note saying "hey, see how this trait means you don't need to do a redundant check."

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1141/commits/6b6976c97d5f7ddc45d17ddac628ffa696e5d3b3

Have just added the publishing check, see comment on the above example. If we keep the "Hey, note how we don't use capabilities" example there, I don't think we need to keep repeating ourselves and put it here as well.

Reply:
@feltech You happy with omitting here?

Reply:
You mean omitting `managementPolicy`? Fine by me

Reply:
Ah I think I get you now, as in no need to add `hasCapability(kResolution)` because `ResolvesFutureEntities` implies it?  Yeah that's fine.